src/abstractdataframe/show.jl

```diff
@@ -315,21 +317,30 @@ function showrows(io::IO,
                   rowindices2::AbstractVector{Int},
                   maxwidths::Vector{Int},
                   splitchunks::Bool = false,
+                  allcols::Bool = true,
                   rowlabel::Symbol = :Row,
                   displaysummary::Bool = true) # -> Void
     ncols = size(df, 2)
 
-    if displaysummary
-        println(io, summary(df))
-    end
-
     if isempty(rowindices1)
+        if displaysummary
+            println(io, summary(df))
+        end
         return
     end
 
     rowmaxwidth = maxwidths[ncols + 1]
     chunkbounds = getchunkbounds(maxwidths, splitchunks, displaysize(io)[2])
     nchunks = length(chunkbounds) - 1
```

Comment:
Would be clearer to do `nchunks = allcols ? length(chunkbounds) - 1 : min(nchunks, 1)`.

Reply:
changed (with a bit different code as `nchunks` is undefined before this line)